src/snowflake/connector/s3_storage_client.py

```diff
@@ -101,6 +101,10 @@ def __init__(
                     f"https://{self.s3location.bucket_name}.s3.amazonaws.com"
                 )
 
+    @staticmethod
+    def _url_quote(url: str) -> str:
```

Comment:
Add a doc-string that describes why this is necessary

Reply:
It looks like this is only an issue on Python 3.6

> Python 3.7 updates from using RFC 2396 to RFC 3986 to quote URL strings. Now, "~" is included in the set of unreserved characters.

Make sure that you add this to the doc-string.